seps/1933-workload-identity-federation.md

```diff
@@ -0,0 +1,228 @@
+
+### Design Decisions
+
+**Why JWT Bearer Grant Type?**
+
+The JWT Bearer grant type (RFC 7523) is a well-established OAuth 2.0 extension specifically designed for workload-to-service authentication scenarios. It provides:
+
+- Standardized protocol
+- Clear validation requirements
+- Broad implementation support across authorization servers
+- Natural alignment with existing workload identity systems
+
+**Why OpenID Connect Discovery?**
+
+OpenID Connect Discovery provides a standardized mechanism for:
+
+- Automatic discovery of issuer configuration
+- Dynamic retrieval of signing keys via `jwks_uri`
+- Reduced manual configuration burden
+- Support for key rotation without service disruption
+
+This automation is critical for operational simplicity in dynamic workload environments where issuers may rotate keys frequently.
+
+### Alternatives Considered
+
+**Client Credentials Flow with Dynamic Registration**
+
+We considered extending the Client Credentials extension with dynamic client registration. This was rejected because:
+
+- It still requires managing per-client credentials (client secrets or private keys)
+- Dynamic registration adds complexity without solving the credential management problem
+- Workloads already have cryptographically verifiable identities that would be unused
+
+## Backward Compatibility
+
+There are no backward compatibility concerns. This SEP introduces a new optional authorization extension and does not modify existing MCP authorization mechanisms. Implementation of this extension is entirely optional. Systems not requiring workload identity federation can continue using existing authorization mechanisms without modification.
+
+## Security Implications
+
+Workload Identity Federation relies on **platform-provisioned, short-lived workload credentials** (such as Kubernetes service account tokens, SPIFFE SVID-JWTs, or cloud-issued workload tokens). These credentials typically have strong issuance guarantees and tight lifetimes, reducing the window in which a compromised token can be misused. Because these JWTs act as authorization grants, implementations MUST treat them as highly sensitive.
+
+When exchanging tokens, both MCP clients and authorization servers MUST ensure that workload JWTs are handled securely:
+
+- Workload JWTs MUST be transmitted only over authenticated, TLS-protected channels.
+- Implementations SHOULD avoid logging, storing, or otherwise persisting either JWT assertions or exchanged access tokens.
+- Authorization servers SHOULD verify token freshness (`exp`, `iat`, `jti`) and enforce strict replay protection.
```

Comment:
Same as above, this feels like an implementation detail that should be captured by an OAuth RFC.